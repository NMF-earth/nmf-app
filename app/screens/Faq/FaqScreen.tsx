import React from "react";
import { ScrollView } from "react-native";

import Accordion from "components/Accordion";
import { NavStatelessComponent } from "interfaces";

import contents from "./contents";
import styles from "./styles";

const FaqScreen: NavStatelessComponent = () => {
  return (
    <ScrollView style={styles.container}>
      <Accordion>
        {contents.map((content) => (
          <Accordion.Item key={content.key} title={content.title}>
            {content.description}
          </Accordion.Item>
        ))}
      </Accordion>
    </ScrollView>
  );
};

export default FaqScreen;
