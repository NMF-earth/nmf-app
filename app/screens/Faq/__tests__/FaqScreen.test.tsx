import React from "react";
import { create } from "react-test-renderer";

import FaqScreen from "../FaqScreen";

it("FaqScreen renders correctly", () => {
  const tree = create(<FaqScreen />).toJSON();

  expect(tree).toMatchSnapshot();
});
