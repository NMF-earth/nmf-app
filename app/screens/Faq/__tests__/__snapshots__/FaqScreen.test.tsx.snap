// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "paddingHorizontal": 16,
      "paddingVertical": 24,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E7F3EB",
              "borderRadius": 10,
              "borderWidth": 1.5,
              "marginTop": 16,
            },
            Object {
              "marginTop": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 16,
            }
          }
        >
          <Ionicons
            name="chevron-up"
            size={22}
            style={
              Object {
                "color": "#3957D0",
              }
            }
          />
          <Text.H3
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "fontSize": 16,
                "marginLeft": 8,
              }
            }
          >
            Why do I have to enter manually my emissions?
          </Text.H3>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E7F3EB",
              "borderRadius": 10,
              "borderWidth": 1.5,
              "marginTop": 16,
            },
            false,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 16,
            }
          }
        >
          <Ionicons
            name="chevron-up"
            size={22}
            style={
              Object {
                "color": "#3957D0",
              }
            }
          />
          <Text.H3
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "fontSize": 16,
                "marginLeft": 8,
              }
            }
          >
            How accurate are my carbon emissions?
          </Text.H3>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E7F3EB",
              "borderRadius": 10,
              "borderWidth": 1.5,
              "marginTop": 16,
            },
            false,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 16,
            }
          }
        >
          <Ionicons
            name="chevron-up"
            size={22}
            style={
              Object {
                "color": "#3957D0",
              }
            }
          />
          <Text.H3
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "fontSize": 16,
                "marginLeft": 8,
              }
            }
          >
            What is your goal?
          </Text.H3>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E7F3EB",
              "borderRadius": 10,
              "borderWidth": 1.5,
              "marginTop": 16,
            },
            false,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 16,
            }
          }
        >
          <Ionicons
            name="chevron-up"
            size={22}
            style={
              Object {
                "color": "#3957D0",
              }
            }
          />
          <Text.H3
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "fontSize": 16,
                "marginLeft": 8,
              }
            }
          >
            Are you a startup?
          </Text.H3>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E7F3EB",
              "borderRadius": 10,
              "borderWidth": 1.5,
              "marginTop": 16,
            },
            false,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 16,
            }
          }
        >
          <Ionicons
            name="chevron-up"
            size={22}
            style={
              Object {
                "color": "#3957D0",
              }
            }
          />
          <Text.H3
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "fontSize": 16,
                "marginLeft": 8,
              }
            }
          >
            Are you a business?
          </Text.H3>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
